rust/dyn-ddlog/src/ast/rewrites/nesting_eliminator.rs

```diff
@@ -0,0 +1,30 @@
+use super::ExpressionRewriter;
+use crate::ast::{Expr, ExprKind};
+use std::mem;
+
+/// Completely removes all `Nested` expressions since they're redundant in the ast
```

Comment:
What are they doing in the AST in the first place then?  Do you keep them for printing purposes?

Reply:
Also, why is there both simplifier and eliminator?